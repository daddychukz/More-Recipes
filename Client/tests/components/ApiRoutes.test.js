import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';

import Routes from '../../src/routes';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const store = mockStore({});

const mountedWrapper = mount(
  <BrowserRouter>
    <Provider store={store}>
      <Routes />
    </Provider>
  </BrowserRouter>
);

describe('The Routes Component', () => {
  it('should render without issues', () => {
    expect(mountedWrapper).toBeDefined();
    expect(toJson(mountedWrapper)).toMatchSnapshot();
  });
});
