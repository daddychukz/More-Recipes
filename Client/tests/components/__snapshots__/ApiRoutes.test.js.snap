// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Routes Component should render without issues 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <myRoutes>
        <BrowserRouter>
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "blank",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <Switch>
              <Route
                component={[Function]}
                computedMatch={
                  Object {
                    "isExact": false,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "blank",
                    "search": "",
                    "state": undefined,
                  }
                }
              >
                <NotFound
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "blank",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "blank",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <div>
                    <Connect(Header)>
                      <Header
                        logout={[Function]}
                      >
                        <div>
                          <nav
                            className="navbar navbar-expand-md bg-dark navbar-dark fixed-top"
                          >
                            <div
                              className="container"
                            >
                              <Link
                                className="navbar-brand"
                                replace={false}
                                to="/recipe-box"
                              >
                                <a
                                  className="navbar-brand"
                                  href="/recipe-box"
                                  onClick={[Function]}
                                >
                                  <h3
                                    id="logo"
                                  >
                                    More-Recipes
                                  </h3>
                                </a>
                              </Link>
                              <button
                                className="navbar-toggler"
                                data-target="#navbarCollapse"
                                data-toggle="collapse"
                              >
                                <span
                                  className="navbar-toggler-icon"
                                />
                              </button>
                              <div
                                className="collapse navbar-collapse"
                                id="navbarCollapse"
                              >
                                <ul
                                  className="navbar-nav ml-auto"
                                >
                                  <li
                                    className="nav-item"
                                  >
                                    <Link
                                      className="nav-link false"
                                      id="home"
                                      replace={false}
                                      to="/recipe-box"
                                    >
                                      <a
                                        className="nav-link false"
                                        href="/recipe-box"
                                        id="home"
                                        onClick={[Function]}
                                      >
                                        Home
                                      </a>
                                    </Link>
                                  </li>
                                  <li
                                    className="nav-item"
                                  >
                                    <Link
                                      className="nav-link"
                                      id="addRecipe"
                                      replace={false}
                                      to="/add-recipe"
                                    >
                                      <a
                                        className="nav-link"
                                        href="/add-recipe"
                                        id="addRecipe"
                                        onClick={[Function]}
                                      >
                                        Add Recipe
                                      </a>
                                    </Link>
                                  </li>
                                  <li
                                    className="nav-item"
                                  >
                                    <Link
                                      className="nav-link"
                                      id="my-recipe"
                                      replace={false}
                                      to="/my-recipe"
                                    >
                                      <a
                                        className="nav-link"
                                        href="/my-recipe"
                                        id="my-recipe"
                                        onClick={[Function]}
                                      >
                                        My Recipes
                                      </a>
                                    </Link>
                                  </li>
                                  <li
                                    className="nav-item"
                                  >
                                    <Link
                                      className="nav-link"
                                      replace={false}
                                      to="/my-favorite"
                                    >
                                      <a
                                        className="nav-link"
                                        href="/my-favorite"
                                        onClick={[Function]}
                                      >
                                        Favorites
                                      </a>
                                    </Link>
                                  </li>
                                  <li
                                    className="nav-item"
                                  >
                                    <div
                                      className="btn-group open"
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="fa fa-user-circle fa-2x pull-right"
                                        data-toggle="dropdown"
                                        id="drop-down"
                                      />
                                      <ul
                                        className="dropdown-menu"
                                      >
                                        <li>
                                          <Link
                                            className="dropdown-item"
                                            replace={false}
                                            to="/my-profile"
                                          >
                                            <a
                                              className="dropdown-item"
                                              href="/my-profile"
                                              onClick={[Function]}
                                            >
                                              Profile
                                            </a>
                                          </Link>
                                        </li>
                                        <li
                                          className="divider"
                                        />
                                        <li>
                                          <Link
                                            className="dropdown-item"
                                            id="logout"
                                            onClick={[Function]}
                                            replace={false}
                                            to="/"
                                          >
                                            <a
                                              className="dropdown-item"
                                              href="/"
                                              id="logout"
                                              onClick={[Function]}
                                            >
                                              Logout
                                            </a>
                                          </Link>
                                        </li>
                                      </ul>
                                    </div>
                                  </li>
                                </ul>
                              </div>
                            </div>
                          </nav>
                          <div
                            className="modal fade text-dark"
                            id="contactModal"
                          >
                            <div
                              className="modal-dialog"
                            >
                              <div
                                className="modal-content"
                              >
                                <div
                                  className="modal-header bg-primary"
                                >
                                  <h5
                                    className="modal-title"
                                    id="contactModalTitle"
                                    style={
                                      Object {
                                        "color": "white",
                                      }
                                    }
                                  >
                                    Contact Us
                                  </h5>
                                </div>
                                <div
                                  className="modal-body"
                                >
                                  <form>
                                    <div
                                      className="form-group"
                                    >
                                      <label
                                        htmlFor="name"
                                      >
                                        Name
                                      </label>
                                      <input
                                        className="form-control"
                                        type="text"
                                      />
                                    </div>
                                    <div
                                      className="form-group"
                                    >
                                      <label
                                        htmlFor="email"
                                      >
                                        Email
                                      </label>
                                      <input
                                        className="form-control"
                                        type="email"
                                      />
                                    </div>
                                    <div
                                      className="form-group"
                                    >
                                      <label
                                        htmlFor="messager"
                                      >
                                        Message
                                      </label>
                                      <textarea
                                        className="form-control"
                                      />
                                    </div>
                                  </form>
                                </div>
                                <div
                                  className="modal-footer"
                                >
                                  <button
                                    className="btn btn-primary btn-block"
                                  >
                                    Submit
                                  </button>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </Header>
                    </Connect(Header)>
                    <header
                      id="header"
                    />
                    <section>
                      <div
                        className="container"
                      >
                        <div
                          className="row"
                        >
                          <div
                            className="col-6 obj-color"
                          >
                            <h2>
                              404! Resource Not Found
                            </h2>
                          </div>
                        </div>
                      </div>
                    </section>
                  </div>
                </NotFound>
              </Route>
            </Switch>
          </Router>
        </BrowserRouter>
      </myRoutes>
    </Provider>
  </Router>
</BrowserRouter>
`;
